subprojects/code-quality/src/integTest/groovy/org/gradle/api/plugins/quality/PmdPluginBuildCacheIntegrationTest.groovy

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins.quality
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.DirectoryBuildCacheFixture
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
+
+import static org.hamcrest.Matchers.containsString
+import static org.hamcrest.Matchers.not
+
+class PmdPluginBuildCacheIntegrationTest extends AbstractIntegrationSpec implements DirectoryBuildCacheFixture {
+
+    def setup() {
+        buildFile << """
+            apply plugin: "java"
+            apply plugin: "pmd"
+            pmd {
+                incrementalAnalysis = true
+            }
+
+            ${mavenCentralRepository()}
+
+            ${!TestPrecondition.FIX_TO_WORK_ON_JAVA9.fulfilled ? "sourceCompatibility = 1.6" : ""}
```

Comment:
❌ Why do we need this line? If you really want to do something conditionally for running on Java 8 or earlier, please use `JDK8_OR_EARLIER`. `FIX_TO_WORK_ON_JAVA9` means we still need to do some work to make whatever happens to work on Java 9 and therefore we are ignoring the test.
I don't see why you can't set source compatibility to `1.7` if you really need to or not all.

Reply:
Thanks, I was just copying from previous integration tests. I've removed it now.